Mathlib/FieldTheory/Relrank.lean

```diff
@@ -0,0 +1,530 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.FieldTheory.Adjoin
+
+/-!
+
+# Relative rank of subfields and intermediate fields
+
+This file contains basics about the relative rank of subfields and intermediate fields.
+
+## Main definitions
+
+- `Subfield.relrank A B`, `IntermediateField.relrank A B`:
+  defined to be `[B : A ⊓ B]` as a `Cardinal`.
+  In particular, when `A ≤ B` it is `[B : A]`, the degree of the field extension `B / A`.
+  This is similar to `Subgroup.relindex` but it is `Cardinal` valued.
+
+- `Subfield.relfinrank A B`, `IntermediateField.relfinrank A B`:
```

Comment:
Should these be `relRank` and `relFinrank`? I think the group theory one should be `relIndex` too. And then the file name would be RelRank ...

Reply:
I have no preferences on this. Maybe you can open a poll on zulip...

Reply:
Let's not worry about this here.